### ISSUE TITLE
 EventGetter.max_events and broken files

### ISSUE THREAD

**Issue by dneise** [cta-observatory/dragonboard_testbench#32]
Luckily we got a broken file,  so we can also try to write the EventGetter such that working with broken files is convenient.

Until recently I thought raising an exception, when reaching a broken event is the right thing to do, since the file is broken and the user should know about it. Now I think, well yes, but the reaction of the user is:
 * Hell yeah, the file might be broken, so don't return the last broken event, but iterate until the last good event, but don't bother me about it.

So now I think the EventGetter should issue a [warning](https://docs.python.org/3.1/library/warnings.html) when opening such a file. 

Why already when opening? Well for the new `__repr__` feature (see also #31) , we calculate the number of events in a file already at this early stage. So we already see very early that the number of events is not an integer.

I also think, that based on this calculation of the number of events in the file, the member `max_events` should be set to this number. 

Then while iterating we can use `max_events` in order to stop before the last broken event and not bother the user with any exceptions at this stage.

**Comment by dneise** [comment]
I made a test implementation which behaves like this

~~~
In [1]: import dragonboard as db

In [2]: eg = db.EventGenerator("../raw_data_20160229/20160229_185742_ExtTrgRand/Ped_Rand1kHz_No5_IP90.dat")
/home/dneise/LST/dbtb_new/dragonboard/io.py:94: UserWarning: 
File:
/home/dneise/LST/raw_data_20160229/20160229_185742_ExtTrgRand/Ped_Rand1kHz_No5_IP90.dat
might be broken.
Number of events is not integer but 18360.70.
  num_events

In [3]: eg
Out[3]: 
EventGenerator_v5_1_0B(
path='/home/dneise/LST/raw_data_20160229/20160229_185742_ExtTrgRand/Ped_Rand1kHz_No5_IP90.dat', 
max_events=18360)
roi ......: 40
event_size: 1344 byte
#events ..: 18360

In [4]: for event in eg:
   ...:     pass
   ...: 

In [5]: event.header.event_counter
Out[5]: 18360

~~~

**Comment by dneise** [comment]
Also iterating over the EventGetter with `twdm` looks now nice:
~~~
In [1]: import dragonboard as db

In [2]: from tqdm import tqdm

In [3]: eg = db.EventGenerator("../raw_data_20160229/20160229_185742_ExtTrgRand/Ped_Rand1kHz_No5_IP90.dat")
/home/dneise/LST/dbtb_new/dragonboard/io.py:94: UserWarning: 
File:
/home/dneise/LST/raw_data_20160229/20160229_185742_ExtTrgRand/Ped_Rand1kHz_No5_IP90.dat
might be broken.
Number of events is not integer but 18360.70.
  num_events

In [4]: for event in tqdm(eg, leave=True):
    pass
   ...: 
100%|██████████████████████████████████████████████████████████████████████| 18360/18360 [00:02<00:00, 9018.48it/s]

~~~

**Comment by MaxNoe** [comment]
This was fixed by merging #33
